Sources/NIO/Socket.swift

```diff
@@ -169,7 +186,11 @@ typealias IOVector = iovec
                  controlBytes: UnsafeMutableRawBufferPointer) throws -> IOResult<Int> {
         // Dubious const casts - it should be OK as there is no reason why this should get mutated
         // just bad const declaration below us.
-        var vec = iovec(iov_base: UnsafeMutableRawPointer(mutating: pointer.baseAddress!), iov_len: pointer.count)
+#if os(Windows)
+        var vec = IOVector(len: ULONG(pointer.count), buf: UnsafeMutablePointer<CHAR>(mutating: pointer.baseAddress!.assumingMemoryBound(to: CHAR.self)))
```

Comment:
Ok, there are a lot of assumptions about the memory binding here. Is there any reason not to just change the type of the base pointer here on Windows?

Reply:
I was trying to avoid two entire paths, but we can do that.

Reply:
Do we get two entire paths? I think we can typealias the actual type and use that type everywhere.

Reply:
Ah, let me try that approach - using a typealias instead would certainly be better.

Reply:
I opted to not do this since re-working this made this the same as every other instance.